source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -642,152 +652,377 @@ namespace aspect
 
         // create shortcuts
         const std::array<double, 4> &tau = ref_resolved_shear_stress;
-
         std::vector<double> strain_energy(n_grains);
         double mean_strain_energy = 0;
-
-        for (unsigned int grain_i = 0; grain_i < n_grains; ++grain_i)
+        //debugging init
+        // if (deformation_type== DeformationType::olivine_d_fabric)
+        //   {
+        //     std::cout << "-------------------" <<std::endl;
+        //     std::cout << "\n def_type=olivined" <<std::endl;
+        //   }
+        // else if (deformation_type== DeformationType::clinopyroxene)
+        //   {
+        //     std::cout << "def_type=CPX" <<std::endl;
+        //   }
+
+        // for (int i = 3; i >= 0; i--)
+        //   std::cout << "outRRSS["<<i<<"]"<< ref_resolved_shear_stress[i]<<std::endl;
+        //debugging ends
+
+        // CPX and olivine_d_fabric has different slip systems
+        // // first initiate the l and n
+        // std::array<Tensor<1,3>,4> slip_normal_reference {{Tensor<1,3>({0,1,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,1,0}),Tensor<1,3>({1,0,0})}};
+        // std::array<Tensor<1,3>,4> slip_direction_reference {{Tensor<1,3>({1,0,0}),Tensor<1,3>({1,0,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,0,1})}};
```

Comment:
Remove all commented code. For a reader it is never possible to understand why this code is commented.